docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,285 @@
+Remote Attestation Collaterals
+=====
+
+Adding attestation APIs to support collaterals for remote attestation.  Collaterals are information from a second source that the user can provide for attestation verification.
+
+Motivation
+----------
+
+Currently, the existing collaterals used for Intel SGX quote verification are not exposed to the user.  Adding these new APIs allows the user to specify these collaterals instead of using the default collaterals used by OE SDK.
+
+
+User Experience
+---------------
+
+There are 3 user scenarios.
+
+### 1. User does not specify the collaterals:
+
+##### Server generates the evidence (inside an enclave)
+```C
+// Get OE report
+result = oe_get_report(
+    OE_REPORT_FLAGS_REMOTE_ATTESTATION,
+    NULL, // report_data
+    0,
+    NULL, // opt_params
+    0,
+    &report,
+    report_size);
+```
+
+##### Challenger verifies the evidence (in the host)
+```C
+...
+
+// Verify report without collaterals
+result = oe_verify_remote_report_with_collaterals(
+            report,
+            report_size,
+            NULL, // collaterals
+            0,    // collaterals_size
+            NULL, // input_validation_time
+            parsed_report);
+...
+```
+
+### 2. User is provided with collaterals:
+
+##### Server generates the evidence and collaterals (inside an enclave)
+```C
```

Comment:
What do you mean by "server" here?

Reply:
The server implies a machine that supports a TEE.

Reply:
I see. In that case, it’s better to use the attester instead of server.